### PR TITLE
C6717 added

### PR DESCRIPTION
**Pull request by KyryloBrener**

### CODE REVIEW

File: cypress/support/fragments/orders/orders.js [hunk 1]
```diff
@@ -131,5 +131,28 @@ export default {
 
   closeThirdPane: () => {
     cy.do(PaneHeader({ id: 'paneHeaderorder-details' }).find(Button({ icon: 'times' })).click());
-  }
+  },
+
+  getSearchParamsMap(orderNumber, currentDate) {
```
Comment (dmitriyfolio) [inline cypress/support/fragments/orders/orders.js:136]: there is a list of parameters which should be checked:
<li>Keyword</li>
<li>Date created</li>
<li>Date opened</li>
<li>PO number</li>
Look's like "Date opened" should be added, but see you've already have such check into another method. Add comment about absence of Date opened.

File: cypress/support/fragments/orders/orders.js [hunk 2]
```diff
@@ -131,5 +131,28 @@ export default {
...
+    for (const [key, value] of searchParamsMap.entries()) {
+      cy.do([
+        SearchField({ id: 'input-record-search' }).selectIndex(key),
+        SearchField({ id: 'input-record-search' }).fillIn(value),
+        Button('Search').click(),
+      ]);
+      // verify that first row in the result list contains related order line title
+      this.checkSearchResults(orderNumber);
+      this.resetFilters();
+      // TODO: remove waiter - currenty it's a workaround for incorrect selection from search list
```
Comment (dmitriyfolio) [inline cypress/support/fragments/orders/orders.js:154]: if it's related wit folio bug - add link to the ticket in Jira

File: cypress/integration/orders/orders.PoSearch.spec.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    Orders.createOrderWithOrderLineViaApi(order, orderLine)
+      .then(orderNumber => {
+        const today = new Date();
+        cy.visit(TopMenu.ordersPath);
+        Orders.checkPoSearch(Orders.getSearchParamsMap(orderNumber, DateTools.getFormattedDate({ date: today }, 'MM/DD/YYYY')),
+          orderNumber);
+        // open order to check 'date opened' search
+        Orders.searchByParameter('PO number', orderNumber);
+        SearchHelper.selectFromResultsList();
+        Orders.openOrderViaActions();
```
Comment (dmitriyfolio) [inline cypress/integration/orders/orders.PoSearch.spec.js:44]: do we have several ways to open order? If no - le't rename  Orders.openOrderViaActions(); to  Orders.openOrder();

File: cypress/support/fragments/orders/orders.js [hunk 4]
```diff
@@ -131,5 +131,28 @@ export default {
...
+    searchParamsMap.set('PO number', orderNumber)
+      .set('Keyword', orderNumber)
+      .set('Date created', currentDate);
+    return searchParamsMap;
+  },
+
+  checkPoSearch(searchParamsMap, orderNumber) {
+    for (const [key, value] of searchParamsMap.entries()) {
+      cy.do([
+        SearchField({ id: 'input-record-search' }).selectIndex(key),
```
Comment (dmitriyfolio) [inline cypress/support/fragments/orders/orders.js:147]: move  SearchField({ id: 'input-record-search' } to constant

File: cypress/integration/orders/orders.PoSearch.spec.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import TestType from '../../support/dictionary/testTypes';
+import Orders from '../../support/fragments/orders/orders';
+import TopMenu from '../../support/fragments/topMenu';
+import DateTools from '../../support/utils/dateTools';
+import SearchHelper from '../../support/fragments/finance/financeHelper';
+
+describe('orders: Test PO search', () => {
+  const order = { ...NewOrder.defaultOrder };
+  const orderLine = { ...BasicOrderLine.defaultOrderLine };
+  const locationName = 'Main Library';
```
Comment (dmitriyfolio) [inline cypress/integration/orders/orders.PoSearch.spec.js:12]: there are several places with Main Library. Let's try to move it into one place
image

File: cypress/support/fragments/orders/orders.js (hunk 1 above)
Comment (KyryloBrener) [reply cypress/support/fragments/orders/orders.js:137]: Updated

File: cypress/integration/orders/orders.PoSearch.spec.js (hunk 3 above)
Comment (KyryloBrener) [reply cypress/integration/orders/orders.PoSearch.spec.js]: Updated

File: cypress/support/fragments/orders/orders.js (hunk 4 above)
Comment (KyryloBrener) [reply cypress/support/fragments/orders/orders.js]: Updated

File: cypress/integration/orders/orders.PoSearch.spec.js (hunk 5 above)
Comment (KyryloBrener) [reply cypress/integration/orders/orders.PoSearch.spec.js]: Updated

**Review by MaksimTerekhov — APPROVED** [review]

**Pull request merged**